packages/aragon-wrapper/src/identity/AddressBookIdentityProvider.js

```diff
@@ -0,0 +1,64 @@
+import AddressIdentityProvider from './AddressIdentityProvider'
+import { first, map } from 'rxjs/operators'
+import { getCacheKey } from '../utils/index'
+
+const addressBookAppIds = [
+  '0x32ec8cc9f3136797e0ae30e7bf3740905b0417b81ff6d4a74f6100f9037425de'
+  // TODO Add in App Ids for rinkeby and mainnet appIds
+]
+/**
+ * An identity provider for Address Book Entries
+ *
+ * @class AddressIdentityProvider
+ */
+export default class AddressBookIdentityProvider extends AddressIdentityProvider {
+  constructor (apps, cache) {
+    super()
+    this.apps = apps
+    this.cache = cache
+  }
+  /**
+   * Optional initialization, if required by the provider
```

Comment:
And actually, this is the code that calls `init`:

```js
      // Most providers should have this defined to a noop function by default, but just in case
      if (typeof provider.init === 'function') {
        return provider.init()
      }
```

Given that it checks if `init` exists before trying to call it, can't we omit the function altogether?

Reply:
If we omit it entirely, this class will inherit the no-op `init` from `AddressIdentityProvider`. I think that seems better.

I doubt it makes sense to have that `if (typeof ...)` check; I'd prefer the interface of identity providers were strictly enforced, but that's a separate conversation!